src/libhirte/common/cfg.c

```diff
@@ -137,6 +141,29 @@ int cfg_load_from_file(struct config *config, const char *config_file) {
         return ini_parse(config_file, parsing_handler, config);
 }
 
+char *is_file_name_ending_with_conf(char *file_name, char *conf_file_enfing) {
+        return strstr(file_name, conf_file_enfing);
+}
+
+int cfg_load_from_dir(struct config *config, const char *custom_config_directory) {
+        struct dirent **all_files_in_dir = NULL;
+        int number_of_files = 0;
+        char *conf_file_enfing = ".conf";
+
+        number_of_files = scandir(custom_config_directory, &all_files_in_dir, NULL, alphasort);
+        if (number_of_files == -1) {
+                return ENOENT;
+        }
```

Comment:
```suggestion
        number_of_files = scandir(custom_config_directory, &all_files_in_dir, NULL, alphasort);
        if (number_of_files < 0) {
                return -errno;
        }
```
According to man page (see [here](https://man7.org/linux/man-pages/man3/scandir.3.html)), if `scandir` fails, `errno` will be set - which then contains the last error that occurred. Since `scandir` could also fail due to other reasons (e.g. `ENOTDIR`), we should return it.